stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -268,6 +268,26 @@ lazy_static! {
     ADD COLUMN height_in_tenure;
     "#.into(),
     ];
+
+    pub static ref NAKAMOTO_CHAINSTATE_SCHEMA_4: [&'static str; 2] = [
+    r#"
+        UPDATE db_config SET version = "7";
+    "#,
+    // Add a `signer_stats` table to keep track of how many blocks have been signed by each signer
+    r#"
+        -- Table for signer stats
+        CREATE TABLE signer_stats (
+            -- Signers public key
+            public_key TEXT NOT NULL,
+            -- Stacking rewards cycle ID
+            reward_cycle INTEGER NOT NULL,
+            -- Number of blocks signed during reward cycle
+            blocks_signed INTEGER DEFAULT 1 NOT NULL,
```

Comment:
I think you can just have `INTEGER NOT NULL`?

Reply:
Oh I see, you're using `ON CONFLICT` to treat primary key conflicts as a hint to increment `blocks_signed` (which would only happen on the second and subsequent inserts).